### DISCUSSION TITLE
A new GTranslate plugin for e107

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5361] [Show and tell]
# About plugin

This plugin is just my playing with e107 preferences and testing their limits.

It translates e107 with GTranslate plugin, that uses Google Translate automatic translation service to translate e107 site with Google power and make it multilingual.  You can find more information: https://gtranslate.io/

e107 plugin uses a free version only that you can find on the [Website Translator Widget](https://gtranslate.io/website-translator-widget) page.

# Preferences

## **General**

Displayed options depend on selected type of widget.

![image](https://github.com/user-attachments/assets/9b52730a-6d56-407a-b673-ecfbc986c81f)

![image](https://github.com/user-attachments/assets/03bdf2c1-6ae4-4991-96d8-82af94f682a5)

etc.

It is possible that I missed to hide some options, just report this (compare with official widget)

## **CDN / Local Tab**

![image](https://github.com/user-attachments/assets/eb3631b9-1751-43ce-9aea-43d8407f4725)

Url structure is not offered in free services version.

Flags are always loaded from the local folder. 

## **Languages Tab**

**Translate from** is disabled, it is set by Frontend language of your site (value of CORE_LC from languages files)

**Translate to** - select desired translations
Note: if something fails with settings on frontend, all available languages are displayed

**Native Language Names** 
ON:

![image](https://github.com/user-attachments/assets/04a09ad5-5850-4cdd-85f5-d85e637238db)

OFF:

![image](https://github.com/user-attachments/assets/3e7f8ff7-edc1-46a7-892b-9eae6f35d476)

**Auto switch to browser language**: ON/OFF

**Configuration screen:**

![image](https://github.com/user-attachments/assets/a3b6cd86-1205-484c-8975-aa9c3fc7ff32)

# How to use it

- shortcode {GTRANSLATE} in your theme (f.e. theme.html). 

- shortcode menu for inserting this shortcode in Menu manager (not core solution)

- create new Menu and insert content there. Select menu template (core solution). Result depends on your theme.

**Comment by tgtje** [comment]
Interesting, going to take a look at it (just some time).
Pity deepL hasn't such. I like their translations better overall.

**Reply by tgtje** [reply]
@Jimmi08    ![thumbs-up-line-icon](https://github.com/user-attachments/assets/8f69068c-88c5-4cd2-afbf-0d23da01881e)

**Reply by Jimmi08** [reply]
Thanks, I finally learned how to manipulate the Admin UI table with JavaScript...

**Comment by Vodhin** [comment]
Looks great. Do you think you could add "no language names" so it just displays the flags? (I'm guessing that you already did that but not shown)

**Reply by Jimmi08** [reply]
Just select Flags as widget type

![image](https://github.com/user-attachments/assets/0d70a7ca-d903-441c-8fe6-08131578842c)

**Comment by rica-carv** [comment]
> ![image](https://private-user-images.githubusercontent.com/5429548/394746574-9b52730a-6d56-407a-b673-ecfbc986c81f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzODIyMzEsIm5iZiI6MTczNDM4MTkzMSwicGF0aCI6Ii81NDI5NTQ4LzM5NDc0NjU3NC05YjUyNzMwYS02ZDU2LTQwN2EtYjY3My1lY2ZiYzk4NmM4MWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMjA0NTMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTEzZDY5OTI0NGMxZDkyOGU3M2I1YTdkY2Y4ZjQzZDI3MTU3MDFhODIyNjg0MzFmNzk1M2UzZjRkMjY3NmFkZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.WRp61d2Hqa46AjG8iMlV1y-1ZtxO6LJFWk4KLobfCp4)
> 

Justa a remark, Portuguese language doesn't use the Brazil flag.... :grinning::smiley::rofl:

**Reply by tgtje** [reply]
@rica-carv  by language setting (Portu...)
![flag](https://github.com/user-attachments/assets/e713478c-4ff6-4298-b706-cfc08afecd85)

For Port...-Brazil...  in dwf js file line 45  >  `        else if(alt_flags[lang] == 'brazil')
            return flags_location+'pt-br'+flag_ext;`
so maybe you need to set main on (alternative) to Brazil (yes/no setting detect lang ??)

**Comment by Alex-e107nl** [comment]
Very nive @Jimmi08 I tried it and indeed with some delay it works perfect! I only have 1 thing, if you install the plugin test and then remove it as a plugin (uninstall) all flags remain in the media manager without an image, did I do something wrong?
